tests/metatype/ttypetraits.nim

```diff
@@ -58,3 +58,29 @@ var xvar: selectType(string)
 xvar = "proba"
 echo xvar.type.name
 
+
+#----------------------------------------------------
+type
+  AA = distinct seq[int]
+  BB = distinct string
+  CC = distinct int
+  AAA = AA
+
+
+var a2: AAA
+var b2: BB
+var c2: CC
+
+static:
+  doAssert(a2 is distinct)
+  doAssert(b2 is distinct)
+  doAssert(c2 is distinct)
+
+  doAssert($distinctBase(typeof(AA))) == "seq[int]")
```

Comment:
* why `typeof(AA)` when AA is already a type ?
* no need for stringification, instead: doAssert distinctBase(AA) == seq[int]

Reply:
do like to test operation chaining here and I am using `$`. It suits well.
Tests like `distinctBase(AA) is seq[int]` are already done in tsugar.nim, no need to repeat them. 
`tsugar` was ported in this PR to use `distinctBase` from `typetraits` module. 
Though I will change test to use `typeof(variable)` instead, this one wasn't tested

Reply:
> do like to test operation chaining here and I am using $. It suits well.

then maybe do it for 1 test, not all tests; stringification of types is being tested in dedicated tests already;
note that your test is strictly speaking weaker since you could have `$A == $B` without `A is B`

> Tests like distinctBase(AA) is seq[int] are already done in tsugar.nim, no need to repeat them.

I don't follow the logic; currently both typetraits.distinctBase is being tested in tsugar.nim (which has `import sugar except distinctBase` and `import ttypetraits`) and in `ttypetraits`. 
typetraits.distinctBase should be tested in exactly one place (ttypetraits)

Also, are there any semantic differences between `typetraits.distinctBase` and `sugar.distinctBase`? if yes, we should use a different name; if no, `sugar.distinctBase` should be made a deprecated alias for `typetraits.distinctBase`  instead of keeping its implementation, and its test in tsugar.nim should be moved to ttypetraits.nim

Reply:
Bla bla bla. 

In the end `$distinctBase(c) == "xxx"` caught bugs that `distinctBase(c) is XX` couldn't. 
Tested.